person.rb

```diff
@@ -1,30 +1,36 @@
+require 'securerandom'
 require './nameable'
 require './decorator'
+require_relative 'rental'
 
 class Person < Nameable
   attr_reader :id
-  attr_accessor :name, :age
+  attr_accessor :name, :age, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
+    @id = SecureRandom.uuid
     @name = name
     @age = age
     @parent_permission = parent_permission
-    super()
+    @rentals = []
   end
 
-  private
-
-  def of_age?
-    @age >= 18
-  end
-
-  public
-
   def can_use_services?
-    of_age? || @parent_permission
+    of_age? && @parent_permission
   end
 
   def correct_name
     @name
   end
+
+  def add_rental(date, book)
+    Rental.new(date, book, self)
+  end
```

Comment:
- [ ] You need to pay attention to the order of your arguments. You defined the Rental constructor to be in this order: `date`, `person`, `book`. In the same way, for the person class, you should pass your parameters to avoid errors.

  ```
  def add_rental(date, book)
    Rental.new(date, self, book)
  end
  ```